core/src/cluster_slots.rs

```diff
@@ -22,22 +19,26 @@ pub type SlotPubkeys = HashMap<Pubkey, u64>;
 #[derive(Default)]
 pub struct ClusterSlots {
     cluster_slots: RwLock<BTreeMap<Slot, Arc<RwLock<SlotPubkeys>>>>,
-    since: AtomicU64,
     validator_stakes: RwLock<Arc<NodeIdToVoteAccounts>>,
     epoch: RwLock<Option<u64>>,
+    cursor: Mutex<Cursor>,
```

Comment:
Would be nice if we had an `AtomicCursor` as well that supported a backing AtomicU64

Reply:
yeah, that makes sense